app.py

```diff
@@ -1,17 +1,65 @@
+"""
+This script creates and runs the Dash application for the Wesnoth MP Database Dashboard.
+
+Usage:
+
+For development:
+    Run `python app.py` to start the development server.
+
+For production:
+    Run `gunicorn --bind :$PORT app:server` to start the production server, where $PORT specifies the port number.
+
+Note: Gunicorn only runs on Linux/Unix systems. For Windows, use `waitress` instead, or use a Linux container/VM to run Gunicorn.
+"""
+
+import json
 import logging
 import os
 import sys
 
-import dash_bootstrap_components as dbc
 import mariadb
 import plotly.express as px
-from dash import Dash, Input, Output, State, callback, dcc, html
+from dash import Input, Output, State, callback
 from dash.exceptions import PreventUpdate
+from flask_caching import Cache
+
+from layout import create_app
+
+
+app = create_app()
+server = app.server
+
+# Set up Flask-Caching for sharing data between callbacks
+# Refer to the documentation for more information:
+# Dash documentation: https://dash.plotly.com/sharing-data-between-callbacks
+# Flask-Caching documentation: https://flask-caching.readthedocs.io/en/latest/
+CACHE_CONFIG = {
+    "CACHE_TYPE": "SimpleCache",  # SimpleCache stores cached data in memory as a Dictionary.
+}
+cache = Cache()
+cache.init_app(app.server, config=CACHE_CONFIG)
+
+
+@cache.memoize()
+def get_target_table():
+    """Fetches and returns the database table name from the configuration options file."""
+    logging.debug(
+        "get_target_table called"
+    )  # This only gets logged the first time the function is called and proves that memoization is functioning.
+    with open("config.json", "r") as f:
```

Comment:
This opens the config file every time this function is called?

Reply:
only the first time the function is called because of `@cache.memoize()`
Confirmed by the debug log statement.

Reply:
Please now see https://github.com/wesnoth/wesnoth-multiplayer-data-dashboard/pull/10/commits/9bbf9f8d2652b918d58298d1b6aac3f4ff41c97b, the statement has been changed.

Reply:
This seems pretty confusing. I would normally expect loading the config file once at program start into some internal config object and using that throughout the app via a function or whatever.

Anyway if you like it like this that's fine.